website/addons/box/tests/test_views.py

```diff
@@ -0,0 +1,554 @@
+    def test_box_import_user_auth_adds_a_log(self, mock_folder_data, mock_account_info):
+        mock_folder_data.return_value = {
+            'name': 'Camera Uploads',
+            'path_collection': {
+                'entries': [
+                    {'name': 'All Files'}
+                ]
+            }
+        }
+        mock_account_info.return_value = {'display_name': 'Mr. Box'}
+        # Node does not have user settings
+        self.node_settings.user_settings = None
+        self.node_settings.save()
+        url = api_url_for('box_import_user_auth', pid=self.project._primary_key)
+        self.app.put(url, auth=self.user.auth)
+        self.project.reload()
+        self.node_settings.reload()
+        last_log = self.project.logs[-1]
+
+        assert_equal(last_log.action, 'box_node_authorized')
+        log_params = last_log.params
+        assert_equal(log_params['node'], self.project._primary_key)
+        assert_equal(last_log.user, self.user)
+
+    def test_box_get_share_emails(self):
+        # project has some contributors
+        contrib = AuthUserFactory()
+        self.project.add_contributor(contrib, auth=Auth(self.user))
+        self.project.save()
+        url = api_url_for('box_get_share_emails', pid=self.project._primary_key)
+        res = self.app.get(url, auth=self.user.auth)
+        result = res.json['result']
+        assert_equal(result['emails'], [u.username for u in self.project.contributors
+                                        if u != self.user])
+
+    def test_box_get_share_emails_returns_error_if_not_authorizer(self):
+        contrib = AuthUserFactory()
+        contrib.add_addon('box')
+        contrib.save()
+        self.project.add_contributor(contrib, auth=Auth(self.user))
+        self.project.save()
+        url = api_url_for('box_get_share_emails', pid=self.project._primary_key)
+        # Non-authorizing contributor sends request
+        res = self.app.get(url, auth=contrib.auth, expect_errors=True)
+        assert_equal(res.status_code, httplib.FORBIDDEN)
+
+    def test_box_get_share_emails_requires_user_addon(self):
+        # Node doesn't have auth
+        self.node_settings.user_settings = None
+        self.node_settings.save()
+        url = api_url_for('box_get_share_emails', pid=self.project._primary_key)
+        # Non-authorizing contributor sends request
+        res = self.app.get(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, httplib.BAD_REQUEST)
+
+
+class TestFilebrowserViews(BoxAddonTestCase):
+
+    def setUp(self):
+        super(TestFilebrowserViews, self).setUp()
+        self.user.add_addon('box')
+        settings = self.user.get_addon('box')
+        oauth = BoxOAuthSettings(user_id='not none', _access_token='Nah')
+        oauth.save()
+        settings.oauth_settings = oauth
+        settings.save()
+        self.patcher = mock.patch('website.addons.box.model.BoxNodeSettings.folder')
```

Comment:
@mfraezz Related to previous comments, is this patcher necessary if `_folder_data` is set?